sanity-io/sanity
# refactor(sanity): update version id to match `versions.[bundle].[id]`

### Description

- Change `newVersion` to use `versions.[bundle].[id]` vs `[bundle].[id]`
- Change `newVersion` to remove `_version`
- Update `getPublishedId`
- Update places where `_version` was being used
- Add `getVersionId` and `isVersionId`
- Remove `isVersion` parameter from `getPublishedId` (and update each location where it was used)
- Update areas where `getVersionId` / `isVersionId` makes sense to be used
- Update tests

❗ 
- Using an alternative groq to `versionsOf` since that changes hasn't landed in CL yet

#### Dependencies
- waiting for `versionsOf` to work with version.(bundle).(id)
- has _version removal landed?
 
### What to review

- Did I miss anything? 
- Does the structure tool work as you'd expect?
- Does the linking documents in the release tool work as expected?
- Can you edit / see the right documents on the structure tool list?
- Can you add new documents to versions and have those have the right ids?
- Any use cases that are breaking and that weren't caught by the tests / manual test?

### How to test
In the preview you can check the release `New release with a punch` which should have already 2 documents with the new id structure. Feel to create new docs and add them to releases.

❗ **old documents will not appear on lists / will appear disabled / won't be part of the number of the bundles metada**
